components/molecules/VCarouselControls/VCarouselControls.vue

```diff
@@ -1,80 +1,143 @@
+}
+
+function setIndicatorPosition(index: number) {
+    if (!scroll.value || !thumb.value) return
+
+    const percent = index / (props.length - 1)
+    const translate = percent * (scroll.value?.offsetWidth - thumb.value?.getBoundingClientRect().width)
+
+    thumb.value.style.translate = translate + 'px 0 '
+}
+
+function onClick(event: Event) {
+    const direction = (event.currentTarget as HTMLButtonElement).name === 'next' ? 1 : -1
+    index.value = index.value + direction
+}
 </script>
 
 <template>
     <div :class="$style.root">
+        <div
+            ref="scroll"
+            :class="$style.scroll"
+        >
+            <div
+                ref="thumb"
+                :class="$style.thumb"
+            />
+        </div>
         <VButton
-            control
             name="previous"
-            :disabled="index === 0"
+            icon-name="arrow-left"
             :aria-label="$t('carousel.previous_slide_label')"
-            :class="$style.button"
+            :disabled="index === 0"
             :theme="theme"
+            :class="$style.button"
             @click="onClick"
-        >
-            <template #icon>
-                <VIcon name="arrow-left" />
-            </template>
-        </VButton>
+        />
         <div
-            v-if="displayNumbers"
+            v-if="slidePosition"
             class="text-body-xs"
             :class="$style.number"
         >
-            <span>{{ formatValue(index) }} / {{ formatValue(length - 1) }}</span>
+            <span>{{ slidePosition }}</span>
         </div>
         <VButton
-            control
             name="next"
-            :disabled="index === length - 1 || isEnd"
+            icon-name="arrow-right"
             :aria-label="$t('carousel.next_slide_label')"
-            :class="$style.button"
+            :disabled="index === length - 1 || isEnd"
             :theme="theme"
+            :class="$style.button"
             @click="onClick"
-        >
-            <template #icon>
-                <VIcon name="arrow-right" />
-            </template>
-        </VButton>
+        />
     </div>
 </template>
 
 <style lang="scss" module>
 .root {
     display: var(--v-carousel-controls-display, flex);
     align-items: center;
-    justify-content: center;
-    gap: rem(16);
+    justify-content: var(--v-carousel-controls-justify-content, center);
+    gap: rem(8);
+}
 
-    @include media('>=lg') {
-        height: rem(62);
+.scroll {
+    position: relative;
+    overflow: hidden;
+    width: rem(64);
+    height: rem(3);
+    flex-shrink: 0;
+
+    @include media('>=md') {
+        margin-right: rem(40);
+    }
+
+    &::before {
+        position: absolute;
+        background-color: var(--theme-color-controls-selected, color-mix(in srgb, currentColor, transparent 70%));
+        content: '';
+        inset: 0;
+        opacity: 0.2;
     }
 }
 
+.thumb {
+    --thumb-width: 100%;
```

Comment:
is it useful to declare a CSS var here? It couldn't be easily overided. Should we just use the CSS var and let the fallback value if it is not defined?